### PR TITLE
Feature: add modes to create windows in vertical stack

### LINKED ISSUE #725: Shortcut to open new window vertically
**Is your feature request related to a problem? Please describe.**
My most frequently used keyboard shortcut (`super+return`) launches a new terminal window. Most of the time, the default behavior of adding the new window to the right of the currently focused window does what I want but sometimes I want to open the new terminal window _below_ the currently focused window. To do this, I have to execute four keyboard shortcuts:
1. Open terminal (`super+return`)
2. Switch to left window (`super+h` - I remapped)
3. Absorb newly created window (`super+i`)
4. Switch to absorbed window (`super+j` - remapped)

**Describe the solution you'd like**
I used to use i3 and there is a way to modally switch the direction of the next split _prior_ to making the split. So, an idea is to do something like:
1. Switch split direction (`super+v`)
2. Open terminal (`super+return`)

Of course, this would mean having some sort of indication of where the next split would be created would be useful.

### PR DESCRIPTION
**Pull request by jtaala**
This PR improves the situation described in #858, and resolves #725;

More specifically, it adds new "create window" modes:
- `down`: create new windows in vertical stack (below current active window);
- `up`: create new windows in the vertical stack (above current active window);

image

This PR also adds new keybinds (unset by default) to directly set the position to open windows:
image

### CODE REVIEW

File: schemas/org.gnome.shell.extensions.paperwm.gschema.xml [hunk 1]
```diff
@@ -171,6 +171,30 @@
...
+      <summary>Create new windows in vertical stack (up)</summary>
+    </key>
+    <key type="as" name="open-window-position-start">
+      <default><![CDATA[[]]]></default>
+      <summary>Create new windows at start position</summary>
+    </key>
+    <key type="as" name="open-window-position-end">
+      <default><![CDATA[[]]]></default>
+      <summary>Create new windows at end position</summary>
+    </key>
```
Comment (Thesola10) [inline schemas/org.gnome.shell.extensions.paperwm.gschema.xml:197]: Maybe this should be an enum? idk how GNOME does it but `org.gnome.mutter.experimental-features` has the schema I mean

File: tiling.js [hunk 2]
```diff
@@ -3868,15 +3874,35 @@ export function insertWindow(metaWindow, { existing }) {
...
+                stackSlurpTimeout = GLib.timeout_add(
+                    GLib.PRIORITY_DEFAULT,
+                    100,
+                    () => {
+                        slurp(active, SlurpInsertPosition.ABOVE);
+                        return false; // on return false destroys timeout
+                    });
+                break;
+            }
+        });
```
Comment (Thesola10) [inline tiling.js:3905]: Any way to make the animation cleaner? Right now it opens to the side and immediately snaps down as if the keybind was used

**Review by Thesola10 — APPROVED** [review]
:+1:

File: schemas/org.gnome.shell.extensions.paperwm.gschema.xml (hunk 1 above)
Comment (Thesola10) [reply schemas/org.gnome.shell.extensions.paperwm.gschema.xml:197]: In fact reusing the widget that `dconf` uses for that kind of input would be better to tidy up the settings view

Comment (jtaala) [reply schemas/org.gnome.shell.extensions.paperwm.gschema.xml:197]: These are for keybinds... so they can't be enums.

Comment (jtaala) [reply schemas/org.gnome.shell.extensions.paperwm.gschema.xml:197]: Note that the implementation is using an enum:

https://github.com/paperwm/PaperWM/blob/ba27c1bcaf71230911b5183c080b1267132a4dc1/tiling.js#L41

File: tiling.js (hunk 2 above)
Comment (jtaala) [reply tiling.js:3905]: Not currently - the whole slurping and layouting concept is built on the assumption that windows need to be correctly tiled before slurping.

**Pull request merged**